src/base/tests/test_util.py

```diff
@@ -431,6 +431,30 @@ def test_iter_browse_call_twice(self):
 
 
 class TestPG(UnitTestCase):
+    def test_alter_column_type(self):
+        cr = self.env.cr
+        cr.execute(
+            """
+            ALTER TABLE res_partner_title ADD COLUMN x bool;
+
+            UPDATE res_partner_title
+               SET x = CASE id % 3
+                           WHEN 1 THEN NULL
+                           WHEN 2 THEN True
+                           ELSE False
+                       END
+            """
+        )
+        self.assertEqual(util.column_type(cr, "res_partner_title", "x"), "bool")
+        util.alter_column_type(cr, "res_partner_title", "x", "int", using="CASE {0} WHEN True THEN 2 ELSE 1 END")
+        self.assertEqual(util.column_type(cr, "res_partner_title", "x"), "int")
```

Comment:
```suggestion
        self.assertEqual(util.column_type(cr, "res_partner_title", "x"), "int4")
```

Sorry for the first bad proposal.

Reply:
done